build-advisories/src/Roave/SecurityAdvisories/VersionConstraint.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Roave\SecurityAdvisories;
+
+/**
+ * A simple version constraint - naively assumes that it is only about ranges like ">=1.2.3,<4.5.6"
+ */
+final class VersionConstraint
+{
+    const CLOSED_RANGE_MATCHER    = '^>(=?)\s*((\d+.)*\d+)\s*,\s*<(=?)\s*((\d+.)*\d+)$';
+    const LEFT_OPEN_RANGE_MATCHER = '^<(=?)\s*((\d+.)*\d+)$';
+    const RIGHT_OPEN_RANGE_MATCHER = '^>(=?)\s*((\d+.)*\d+)$';
```

Comment:
Yes, but from a VO perspective, it's a valid given constraint. In the bigger picture, they won't be used, but they are still part of the logic of this class